### PR TITLE
Don't require WeakTypeTag for printing (fix #649)

### LINKED ISSUE #649: Wrapper method generated by Ammonite REPL is too large to commit to JVM
The log is too large, see attachment.

[amm-log.txt](https://github.com/lihaoyi/Ammonite/files/1104382/amm-log.txt)

### PR DESCRIPTION
**Pull request by Atry**

### CODE REVIEW

File: amm/repl/src/main/scala/ammonite/repl/FullReplAPI.scala [hunk 1]
```diff
@@ -41,27 +58,11 @@ trait FullReplAPI extends ReplAPI{
...
-      // See https://issues.scala-lang.org/browse/SI-10129 for additional details.
-      val isUnit = try {
-        typeOf[T] =:= typeOf[Unit]
-      } catch {
-        case _: Throwable => value == scala.runtime.BoxedUnit.UNIT
-      }
+    def print[T: pprint.TPrint: FullReplAPI.IsUnit](
+        value: => T,
+        ident: String,
+        custom: Option[String])(implicit tcolors: pprint.TPrintColors) = {
```
Comment (lihaoyi) [inline amm/repl/src/main/scala/ammonite/repl/FullReplAPI.scala]: Let's maintain the old formatting style here

**Comment by lihaoyi** [comment]
The idea looks great. Clearly the tests don't pass now, but If you could make them pass I'll happily merge this

**Comment by Atry** [comment]
I don't know how to fix it. The implicit resolution for `Nothing` seems very strange

**Comment by Atry** [comment]
I simply use `ClassTag` instead of `WeakTypeTag`, as other solutions are all buggy due to https://github.com/scala/bug/issues/10393

**Comment by Atry** [comment]
@lihaoyi How do you think about the new `ClassTag` approach?

**Comment by lihaoyi** [comment]
Looks reasonable. Can you put in a comment that enumerates the obvious/alternative solutions (typetag, special IsUnit typeclass, classtag without the default value) and explains why they dont work in this case. Then I'll merge it since everything is green

**Comment by Atry** [comment]
I didn't mention `IsUnit` type class because this solution is similar to `T =:= Unit`.

**Comment by Atry** [comment]
@lihaoyi Is it mergeable now?

**Comment by lihaoyi** [comment]
Looks good to me

**Pull request merged**